lib/msal-common/src/cache/entities/AccountEntity.ts

```diff
@@ -137,18 +118,25 @@ export class AccountEntity {
      * @param policy
      */
     static createAccount(
-        clientInfo: string,
         homeAccountId: string,
         idToken: AuthToken,
-        authority?: Authority,
+        authority: Authority,
+        clientInfo?: string,
         cloudGraphHostName?: string,
         msGraphHost?: string,
         environment?: string,
         nativeAccountId?: string
     ): AccountEntity {
         const account: AccountEntity = new AccountEntity();
 
-        account.authorityType = CacheAccountType.MSSTS_ACCOUNT_TYPE;
+        if (authority.authorityType === AuthorityType.Adfs) {
```

Comment:
Should we use a map here instead of `if-else` statement?

Reply:
Since we're comparing against both authorityType and protocolMode a map seems trickier to implement here :/